### PR TITLE
New Block: Add "Top Rated Products" block

### LINKED ISSUE #242: Create new block: "Top Rated Products"
See p6riRB-3IN-p2

> The Top Rated Products block inserts a grid of a store’s top rated products.
> A user selects “Top Rated Products” from the block inserter, and the block is placed in the editor. There is no set up screen. 

Sidebar options are Layout & Filter by Category

### PR DESCRIPTION
**Pull request by LevinMedia**
Fixes #242 

This PR adds the new block, "Top Rated Products". 

Top rated shows all products ordered by `rating` in the preview.

You can also filter this block by category, in the block sidebar.

screen shot 2018-12-14 at 1 07 48 pm

**To test**

- Add/edit a post
- Expect: A new block called "Top Rated Products" should appear (can also be found with `woocommerce`)
- Add the block
- Expect: a live preview of your top rated products
- You can edit layout settings, or restrict the category filtering in the sidebar
- Save/publish the post
- View the post
- Expect: same ordering of products on the frontend of the site

### CODE REVIEW

File: assets/js/utils/get-shortcode.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ export default function getShortcode( { attributes }, name ) {
 		case 'woocommerce/product-best-sellers':
 			shortcodeAtts.set( 'best_selling', '1' );
 			break;
+		case 'woocommerce/product-top-rated':
+			shortcodeAtts.set( 'orderby', 'rating', '1' );
```
Comment (ryelle) [inline assets/js/utils/get-shortcode.js]:
```suggestion
			shortcodeAtts.set( 'orderby', 'rating');
```

You don't need the '1' here– this function sets the first param to the second in the shortcode, so this third parameter is not needed. In the other example, the shortcode is `best_selling="1"`, you can also see how we set the orderby a few lines ahead for the blocks that allow you to control that.

**Review by ryelle — COMMENTED** [review]
One tweak, but otherwise this is working great 👍

**Review by ryelle — APPROVED** [review]
There was a mix of tabs & spaces in one file, which the linter doesn't like (you can track that down yourself with `npm run lint`) – I've pushed a fix for that, since it's pretty minor, and this is good to merge 👍

**Comment by LevinMedia** [comment]
>There was a mix of tabs & spaces in one file, which the linter doesn't like (you can track that down yourself with npm run lint)

It's like you read my mind. Thanks for the fix!

**Pull request merged**